### PR TITLE
calculate ticks from ms instead of seconds

### PR DESCRIPTION
**Pull request by spacek531**
I have very little experience with statically-typed languages so YMMV with the `sint32`.

I added the `Min` because `Update()` appears to be incapable of handling negative `_waitCounter` values.

There used to be a `Ceil` function in there but I changed my mind because I realized that it probably casts it to an int.

See also: https://github.com/OpenRCT2/title-sequences/pull/1

### CODE REVIEW

**Comment by spacek531** [comment]
I notice the title sequence pr was merged. Any reason this one has not been merged? I hope this code isn't too awful.

File: src/openrct2/title/TitleSequencePlayer.cpp [hunk 1]
```diff
@@ -247,7 +247,9 @@ class TitleSequencePlayer final : public ITitleSequencePlayer
             _waitCounter = 1;
             break;
         case TITLE_SCRIPT_WAIT:
-            _waitCounter = command->Seconds * UPDATE_FPS;
+
```
Comment (IntelOrca) [inline src/openrct2/title/TitleSequencePlayer.cpp]: Remove empty line

File: src/openrct2/title/TitleSequencePlayer.cpp [hunk 2]
```diff
@@ -247,7 +247,9 @@ class TitleSequencePlayer final : public ITitleSequencePlayer
             _waitCounter = 1;
             break;
         case TITLE_SCRIPT_WAIT:
-            _waitCounter = command->Seconds * UPDATE_FPS;
+		
+			// The waitCounter is measured in 25-ms game ticks. Previously it was seconds * 40 ticks/second, now it is ms / 25 ms/tick
```
Comment (IntelOrca) [inline src/openrct2/title/TitleSequencePlayer.cpp]: Change tabs to spaces

File: src/openrct2/core/Math.hpp [hunk 3]
```diff
@@ -48,4 +48,5 @@ namespace Math
         if (x > 0) return 1;
         return 0;
     }
+
```
Comment (IntelOrca) [inline src/openrct2/core/Math.hpp]: Undo this change.

**Comment by IntelOrca** [comment]
* `TitleCommand.Seconds` needs renaming.
* The string for wait command in seconds needs changing.
* The default value for wait command should be 10000.

**Comment by spacek531** [comment]
I do not know how to say "milliseconds" in all the languages. Hopefully the localisation team can be roused.

File: data/language/en-GB.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# STR_XXXX part is read and XXXX becomes the string id number.
+﻿# STR_XXXX part is read and XXXX becomes the string id number.
```
Comment (Gymnasiast) [inline data/language/en-GB.txt]: MS has the annoying habit of inserting BOMs into UTF-8 files. Please remove it.

**Comment by IntelOrca** [comment]
This will be merged at the same time the title-seqeuences get its new release which includes the new title sequence.

File: src/openrct2/title/TitleSequencePlayer.cpp [hunk 5]
```diff
@@ -247,7 +247,8 @@ class TitleSequencePlayer final : public ITitleSequencePlayer
             _waitCounter = 1;
             break;
         case TITLE_SCRIPT_WAIT:
-            _waitCounter = command->Seconds * UPDATE_FPS;
+            // The waitCounter is measured in 25-ms game ticks. Previously it was seconds * 40 ticks/second, now it is ms / 25 ms/tick
+            _waitCounter = Math::Min<sint32>(1, command->Milliseconds / UPDATE_TIME_MS);
```
Comment (IntelOrca) [inline src/openrct2/title/TitleSequencePlayer.cpp]: This should be `Math::Max`

Comment (spacek531) [inline src/openrct2/title/TitleSequencePlayer.cpp]: I must have had a really bad case of the dumb four days ago.

**Comment by spacek531** [comment]
This will also solve issue https://github.com/OpenRCT2/OpenRCT2/issues/2438

**Comment by spacek531** [comment]
note that unless changing Min to Max magically changes the way things work, this pr does not currently work. I tested the last build and the script did not seem to advance - the camera was stuck in the first position seemingly indefinitely.

**Comment by spacek531** [comment]
I just tested it and changing Min to Max seems to have changed things significantly. Currently copying a script from the ms-updated scripts to test.

**Comment by spacek531** [comment]
I just tested the current sequence with the ms-updated times. Fabulous!

**Comment by spacek531** [comment]
Using the ingame editor for ms is infeasible right now because it is limited to 3 digits, when it should be limited to 5 digits.

**Comment by spacek531** [comment]
sometimes I think the reason github is confusing is because the github gui does nothing but confuse people.

**Comment by spacek531** [comment]
Do we want to merge this sooner and get the localisation team on it before 0.1.0 or get the localisations after 0.1.0?

**Comment by spacek531** [comment]
apparently I can't just change the type in a structure to include more bits.

**Comment by spacek531** [comment]
there is some kind of uint8 bottleneck going both ways between the command list and the text box and I'm out of my depth
image

**Comment by spacek531** [comment]
The movement of the value between file and text box is working fine, but now it seems the number is displaying as a sint16 instead of int16:
image
I looked in here but didn't see anything that jumped out at me as to why.
https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/windows/title_editor.c#L837

**Comment by spacek531** [comment]
No matter what I do I cannot seem to reformat that string using the language files.

**Comment by spacek531** [comment]
I think this is the final version, boys.

At least, I really hope so because once I get the title sequence PR going I'm going to bed.

File: src/openrct2/windows/title_command_editor.c [hunk 6]
```diff
@@ -241,9 +242,11 @@ void window_title_command_editor_open(TitleSequence * sequence, sint32 index, bo
         snprintf(textbox2Buffer, BUF_SIZE, "%d", command.Y);
         break;
     case TITLE_SCRIPT_ROTATE:
```
Comment (janisozaur) [inline src/openrct2/windows/title_command_editor.c]: this should also handle the `ZOOM` case as it did before.

**Comment by spacek531** [comment]
celebration in progress

https://github.com/OpenRCT2/title-sequences/pull/2
